packages/wallet-sdk/src/derive.ts

```diff
@@ -0,0 +1,84 @@
+import { BIP32Interface } from 'bip32';
+import { createSha2Hash, ecPairToHexString } from '@stacks/encryption';
+import { assertIsTruthy } from './utils';
+import { ECPair } from 'bitcoinjs-lib';
+import { Account } from './models/account';
+import { WalletKeys } from './models/wallet';
+
+const DATA_DERIVATION_PATH = `m/888'/0'`;
+const WALLET_CONFIG_PATH = `m/44/5757'/0'/1`;
+const STX_DERIVATION_PATH = `m/44'/5757'/0'/0`;
+
+export const deriveWalletKeys = async (rootNode: BIP32Interface): Promise<WalletKeys> => {
+  assertIsTruthy(rootNode.privateKey);
+  const derived: WalletKeys = {
+    salt: await deriveSalt(rootNode),
+    rootKey: rootNode.toBase58(),
+    configPrivateKey: deriveConfigPrivateKey(rootNode).toString('hex'),
+  };
+  return derived;
+};
+
+/**
+ * Derive the `configPrivateKey` for a wallet.
+ *
+ * This key is derived from the path `m/5757'/0'/1`, using `1` for change option, following the bip44 recommendation
+ * for keys relating to non-public data.
+ *
+ * This key is used to encrypt configuration data related to a wallet, so the user's
+ * configuration can be synced across wallets.
+ *
+ * @param rootNode A keychain that was created using the wallet's seed phrase
+ */
+export const deriveConfigPrivateKey = (rootNode: BIP32Interface): Buffer => {
+  const derivedConfigKey = rootNode.derivePath(WALLET_CONFIG_PATH).privateKey;
+  if (!derivedConfigKey) {
+    throw new TypeError('Unable to derive config key for wallet identities');
+  }
+  return derivedConfigKey;
+};
+
+/**
+ * Generate a salt, which is used for generating an app-specific private key
+ * @param rootNode
+ */
+export const deriveSalt = async (rootNode: BIP32Interface) => {
```

Comment:
Mixing fn expressions and declarations in `/encryption.ts`. For top level fns like this my pref would be function declarations, as should make for a more helpful callstack.

Reply:
I'm not sure I follow this comment. Sorry, probably just the lack of coffee.

Reply:
Oh, you mean `export function x()` vs `export const x = () => {}`? That makes sense, and I like the thought. I tried to be consistent by using destructured options (for more than one arg) and immutability, and this would be another good thing to be consistent about.